File: frontend/src/infrastructure/api/ticketRepository.ts
import type { ITicketRepository } from '../../domains/ticket/repositories/TicketRepository';
import type { Ticket, TicketFilters, CreateTicketParams } from '../../domains/ticket/entities/Ticket';
import apiClient from './apiClient';

export class TicketRepository implements ITicketRepository {
  async getAll(filters?: TicketFilters): Promise<Ticket[]> {
    const params = new URLSearchParams();
    if (filters?.title) params.append('title', filters.title);
    if (filters?.category) params.append('category', filters.category);
    if (filters?.min_price) params.append('min_price', filters.min_price);
    if (filters?.max_price) params.append('max_price', filters.max_price);

    const { data } = await apiClient.get(`/tickets?${params.toString()}`);
    return data;
  }

  async getMyTickets(): Promise<Ticket[]> {
    const { data } = await apiClient.get('/tickets/my');
    return data;
  }

  async create(params: CreateTicketParams): Promise<Ticket> {
    const { data } = await apiClient.post('/tickets', params);
    return data;
  }

  async delete(id: number): Promise<void> {
    await apiClient.delete(`/tickets/${id}`);
  }
}

File: backend/internal/repository/ticket_repository/ticket_repository.go
package ticket_repository

import (
	"context"

	"github.com/TicketX/backend/internal/entity/ticket"
)

// ListTicketParams defines parameters for listing tickets
type ListTicketParams struct {
	MinPrice *float64
	MaxPrice *float64
	Title    *string
	Venue    *string
	Category *string
}

// TicketRepository defines the interface for ticket data operations
type TicketRepository interface {
	Create(ctx context.Context, t *ticket.Ticket) error
	List(ctx context.Context, params *ListTicketParams) ([]*ticket.TicketWithSeller, error)
	GetByID(ctx context.Context, id int) (*ticket.TicketWithSeller, error)
	GetBySellerID(ctx context.Context, sellerID uint) ([]*ticket.Ticket, error)
	Delete(ctx context.Context, id int) error
	FindByID(ctx context.Context, id int) (*ticket.Ticket, error)
	UpdateStatus(ctx context.Context, id int, status ticket.TicketStatus) error
}
